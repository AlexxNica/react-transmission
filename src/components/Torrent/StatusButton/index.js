import React, { Component } from 'react';
import CSSModules from 'react-css-modules';
import { observer } from 'mobx-react';

import styles from './styles/index.css';

@observer
@CSSModules(styles)
class StatusButton extends Component {
  static defaultProps = {
    onToggle: () => {},
  }

  render() {
    return (
      <button
        styleName={this.props.torrent.isStopped ? 'statusResume' : 'statusPause'}
        onClick={this.props.onToggle.bind(this, this.props.torrent.id)}
      />
    );
  }
}

export default StatusButton;
