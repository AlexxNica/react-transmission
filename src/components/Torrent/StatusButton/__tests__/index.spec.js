import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import StatusButton from '../';

test('StatusButton resume', () => {
  const component = renderer.create(
    <StatusButton torrent={{isStopped: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StatusButton pause', () => {
  const component = renderer.create(
    <StatusButton torrent={{isStopped: false}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StatusButton onToggle', () => {
  const toggleStatus = jest.fn();

  const component = shallow(
    <StatusButton torrent={{id: 999, isStopped: false}} onToggle={toggleStatus} />
  );

  component.find('button').simulate('click');

  expect(toggleStatus).toHaveBeenCalledWith(999);
});
