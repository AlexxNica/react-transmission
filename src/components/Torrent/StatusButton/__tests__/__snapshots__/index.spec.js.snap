exports[`test StatusButton pause 1`] = `
<button
  className="statusPause"
  onClick={[Function]} />
`;

exports[`test StatusButton resume 1`] = `
<button
  className="statusResume"
  onClick={[Function]} />
`;
